tasmota/xsns_65_hdc1080.ino

```diff
@@ -0,0 +1,349 @@
+/*
+  xsns_92_hdc1080.ino - Texas Instruments HDC1080 temperature and humidity sensor support for Tasmota
+
+  Copyright (C) 2020  Luis Teixeira
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifdef USE_I2C
+#ifdef USE_HDC1080
+
+/*********************************************************************************************\
+ * HDC1080 - Temperature and Humidy sensor
+ *
+ * Source: Luis Teixeira
+ *
+ * I2C Address: 0x40
+\*********************************************************************************************/
+
+#define XSNS_65             65
+#define XI2C_45             45      // See I2CDEVICES.md
+
+#define HDC1080_ADDR        0x40
+
+// Registers:
+
+#define HDC_REG_TEMP        0x00    // Temperature register
+#define HDC_REG_RH          0x01    // Humidity register
+#define HDC_REG_CONFIG      0x02    // Configuration register
+#define HDC_REG_SERIAL1     0xFB    // First 2 bytes of the serial ID
+#define HDC_REG_SERIAL2     0xFC    // Mid 2 bytes of the serial ID
+#define HDC_REG_SERIAL3     0xFD    // Last bits of the serial ID
+#define HDC_REG_MAN_ID      0xFE    // Manufacturer ID
+#define HDC_REG_DEV_ID      0xFF    // Device ID
+
+// Expected constant values of some of the registers:
+
+#define HDC1080_MAN_ID      0x5449  // Manufacturer ID (Texas Instruments)
+#define HDC1080_DEV_ID      0x1050  // Device ID (valid for the HDC1080)
+
+// Possible values for the configuration register fields:
+
+#define HDC1080_RST_ON      0x8000
+#define HDC1080_HEAT_ON     0x2000
+#define HDC1080_MODE_ON     0x1000  // acquision mode (temperature + humidity)
+#define HDC1080_TRES_11     0x400
+#define HDC1080_HRES_11     0x100
+#define HDC1080_HRES_8      0x80
+
+// Constants:
+
+#define HDC1080_CONV_TIME   80      // Assume 6.50 + 6.35 ms + x of conversion delay for this device
+#define HDC1080_TEMP_MULT   0.0025177
+#define HDC1080_RH_MULT     0.0025177
+#define HDC1080_TEMP_OFFSET 40.0
+
+
+const char* hdc_type_name = "HDC1080";
+uint16_t hdc_manufacturer_id = 0;
+uint16_t hdc_device_id = 0;
+
+float hdc_temperature = 0.0;
+float hdc_humidity = 0.0;
+
+uint8_t hdc_valid = 0;
+
+bool is_reading = false;
+uint32_t timer = millis() + HDC1080_CONV_TIME;
```

Comment:
`timer = millis() + HDC1080_CONV_TIME;` is ran at initialization time of the code, it's not useful here. Just declare `uint32_t timer_hdc1080`. Also it may be wise not to call it just `timer` as this may clash with other global variables.